File: src/medium/reverseInteger.js
/**
 * @copyright https://leetcode.com/problems/reverse-integer/
 * @param {number} num 
 * @returns 
 */
const reverseInteger = (num) => {
	let [result, target]= [num.toString(), ''];
  for (let i = result.length; i > -1; i--) {
    if(!['0', '-'].includes(result.slice(i, i+1))) {
      target += result.slice(i, i+1)
    }
  }
  if (result[0] === '-') target = `-${target}`
return parseInt(target)
}

export default reverseInteger;

File: src/medium/reverse_integer.rs
///@copyright https://leetcode.com/problems/reverse-integer/
/// not solved, has an error.
pub fn reverse_integer(x: i32) -> i32 {
    let temp: String = x.to_string();
    let mut target: String = String::with_capacity(temp.len());
    if temp.contains("-") {
        target.push('-');
    }
    for (pos, i) in temp.bytes().rev().enumerate() {
        if i > 48 && i <= 57 {
            target.push(i as char);
        }
    }
    target.parse().unwrap()
}
#[cfg(test)]
mod tests {
    use super::reverse_integer;
    #[test]
    fn test_reverse_integer() {
        assert_eq!(reverse_integer(123), 321);
        assert_eq!(reverse_integer(-123), -321);
        assert_eq!(reverse_integer(120), 21);
    }
}
